provider/client.go

```diff
@@ -68,8 +102,21 @@ func (client *Client) GetBlock(renterID string, blockID string) (io.ReadCloser,
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
-		resp.Body.Close()
-		return nil, errors.New("bad status code")
+		return nil, decodeError(resp.Body)
+	}
+	return resp.Body, nil
+}
+
+func (client *Client) AuditBlock(renterID string, blockID string) (io.ReadCloser, error) {
```

Comment:
This type signature should be:

```
AuditBlock(renterId, blockID, challenge string) (challengeHash string, err error)
```

It takes a block ID and a challenge string, and returns the hash computed as sha256(challenge + block). 
We can update this later.